odoo/addons/base/models/ir_ui_view.py

```diff
@@ -581,62 +584,84 @@ def _get_filter_xmlid_query(self):
                   WHERE res_id IN %(res_ids)s AND model = 'ir.ui.view' AND module IN %(modules)s
                """
 
-    def get_inheriting_views_arch(self, model):
-        """Retrieves the sets of views that should currently be used in the
-           system in the right order. During the module upgrade phase it
-           may happen that a view is present in the database but the fields it relies on are not
-           fully loaded yet. This method only considers views that belong to modules whose code
-           is already loaded. Custom views defined directly in the database are loaded only
-           after the module initialization phase is completely finished.
-
-           :param str model: model identifier of the inheriting views.
-           :return: list of ir.ui.view
+    def _get_inheriting_views(self):
         """
-        self.ensure_one()
-        self.check_access_rights('read')
-
-        # retrieve all the views transitively inheriting from view_id
-        domain = self._get_inheriting_views_arch_domain(model)
+        Determine the views that inherit from the current recordset, and return
+        them as a recordset, ordered by priority then by id.
+        """
+        domain = self._get_inheriting_views_domain()
         e = expression(domain, self.env['ir.ui.view'])
         from_clause, where_clause, where_params = e.query.get_sql()
-        assert from_clause == '"ir_ui_view"'
-        self.flush(['active'])
-        query = """
+        sub_where_clause = where_clause.replace('ir_ui_view', 'view')
+        assert from_clause == '"ir_ui_view"', f"Unexpected from clause: {from_clause}"
```

Comment:
Ouch :see_no_evil:  Can't you use `where_clause` unmodified in both places inside the full query? The `WITH RECURSIVE` and `UNION` are different scopes so there should be no conflict.